src/Reflection/ReflectionResolver.php

```diff
@@ -82,25 +80,86 @@ public function resolveClassReflection(?Node $node): ?ClassReflection
     public function resolveClassReflectionSourceObject(
         MethodCall|NullsafeMethodCall|StaticCall|PropertyFetch|StaticPropertyFetch $node
     ): ?ClassReflection {
-        if ($node instanceof PropertyFetch || $node instanceof StaticPropertyFetch) {
-            $objectType = $node instanceof PropertyFetch
-                ? $this->nodeTypeResolver->getType($node->var)
-                : $this->nodeTypeResolver->getType($node->class);
+        $objectType = $node instanceof StaticCall || $node instanceof StaticPropertyFetch
+            ? $this->nodeTypeResolver->getType($node->class)
+            : $this->nodeTypeResolver->getType($node->var);
+
+        if (! $objectType instanceof TypeWithClassName) {
```

Comment:
I wonder why we need this checks. these are deprecated in PHPStan.

do you intentionally not use [Type->getObjectClassNames()](https://github.com/phpstan/phpstan-src/blob/89636d6983c73687a9c6d50deac0b01158291f59/src/Type/Type.php#L36) ?

Reply:
or maybe even shortcut to [`Type->getObjectClassReflections`](https://github.com/phpstan/phpstan-src/blob/89636d6983c73687a9c6d50deac0b01158291f59/src/Type/Type.php#L41C18-L41C43)

Reply:
I am not sure, array of objects means can be union that then need to be narrowed or only verify index 0

Reply:
how does the current code currently react in situations like a property fetch which return a native union of `A|B`.

it seems these will silently be skipped?

Reply:
yes, we may need more test for union, eg: falsy object type

- https://github.com/rectorphp/rector-src/pull/6054

Multi object type is skipped, nullable or falsy may need to be checked

Reply:
here can we see a example on how to handle all types - including union - without the need for `instanceof XyType` checks

Reply:
IDE support. The helper method hide the instance type and IDE won't autocomplete its specific methods.

Reply:
calling methods on a Type which is not part of the Type interface is not supported/deprecated.
there should be no need to call a method which only exist for a certain type (or only in very edge cases)